### DISCUSSION TITLE
React Native community package compatibility

### DISCUSSION THREAD

**Discussion by jonstuebe** [facebook/react-strict-dom#68] [Q&A]
What's the idea/story behind how folks can use React Native community packages? Currently, with ﻿react-native-web, most React Native community packages have additional "web" support based on those APIs since they share components like ﻿View, ﻿Text, etc. When transitioning to ﻿react-strict-dom, is the idea for those community packages to switch to using ﻿react-strict-dom for all of their code? Or will there be an abstraction on top of ﻿react-strict-dom that acts as a layer so that folks can use various React Native community packages on the web?

**Comment by necolas** [comment]
Packages rendering UI to web have to use RSD exports, not RN exports

**Comment by LunatiqueCoder** [comment]
Hello. I was wondering what would happen if we use RNfW elements alongside with RSD elements without [`compat`](https://facebook.github.io/react-strict-dom/api/other/compat/) API? I'm mostly worried about gesture handler and reanimated.

I can see in the example app, there's RSD elements nested directly inside a `<ScrollView />` imported from `react-native`:

https://github.com/facebook/react-strict-dom/blob/6dc2762e658d245505c6d57faf6ce3f2ae16aa82/apps/examples/src/components/App.js#L11-L12

https://github.com/facebook/react-strict-dom/blob/6dc2762e658d245505c6d57faf6ce3f2ae16aa82/apps/examples/src/components/App.js#L147-L148

**Comment by MoOx** [comment]
I am thinking about trying RSD on a new project, but yeah I am questioning myself with things like SVG or animation.
I really enjoy react-native-svg and react-native-reanimated even on the web.
For SVG, I guess rn-svg is just throwing svg web components under the hood, so we should be able to use react-native-svg without any issue. But for animation, I am wondering what would be relevant...

**Reply by LunatiqueCoder** [reply]
@MoOx We tried react-native-svg on web and it bundles a lot of useless javascript code. Plain HTML is just better.

**Reply by MoOx** [reply]
Yeah but then your code isnt native compatible 😅

**Reply by LunatiqueCoder** [reply]
Platform specific files do the job for now (.native.ts, .web.ts) and we do our own abtraction, until a better one will be available in the community (perhaps based on RSD)

### EDIT:

React Strict DOM `<svg>` support in progress:
-  https://github.com/facebook/react-strict-dom/pull/272

Reanimated CSS animations:
- https://docs.swmansion.com/react-native-reanimated/docs/next/category/css-animations/
